### PR TITLE
#2288 fixed tab restoration in custom

### LINKED ISSUE #2288: Restoring multiple tabs does not work in feature/navigation custom.
**Describe the bug**

When restarting the custom app, only the most recently opened tab is restored.

**Expected behavior**

All tabs with their y-scroll positions should be restored on app restart. 

**Steps to reproduce the behavior:**
1. Open multiple tabs.
2. Restart the custom app.
3. See that only one tab is restored.

### PR DESCRIPTION
**Pull request by Frans-Lukas**
Fixes #2288 and #2246 once navigation is merged.

Custom now restores tabs on app restart.

### CODE REVIEW

File: .idea/inspectionProfiles/kiwixAndroidInspections.xml [hunk 1]
```diff
@@ -191,7 +187,6 @@
     <inspection_tool class="AndroidLintUnsafeDynamicallyLoadedCode" enabled="true" level="ERROR" enabled_by_default="true" />
     <inspection_tool class="AndroidLintUnsafeNativeCodeLocation" enabled="true" level="ERROR" enabled_by_default="true" />
     <inspection_tool class="AndroidLintUnsafeProtectedBroadcastReceiver" enabled="true" level="ERROR" enabled_by_default="true" />
-    <inspection_tool class="AndroidLintUnsupportedChromeOsCameraSystemFeature" enabled="false" level="ERROR" enabled_by_default="false" />
```
Comment (Frans-Lukas) [inline .idea/inspectionProfiles/kiwixAndroidInspections.xml:194]: hmm, these are odd..

File: custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomReaderFragment.kt [hunk 2]
```diff
@@ -95,22 +95,43 @@ class CustomReaderFragment : CoreReaderFragment() {
       loadUrlWithCurrentWebview(pageUrl)
     } else {
       openObbOrZim()
-      restoreLastOpenedTab()
+      manageExternalLaunchAndRestoringViewState()
```
Comment (macgills) [inline custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomReaderFragment.kt:98]: so this function is in custom and app? probably should be in core then?

Comment (Frans-Lukas) [reply custom/src/main/java/org/kiwix/kiwixmobile/custom/main/CustomReaderFragment.kt:98]: Yes, I think these can be more similar and moved to core.

**Comment by Frans-Lukas** [comment]
I think I need someone to test this on `custom` and `app` because it feels like there are many ways tab restoration can happen. @s-ayush2903 would you mind testing this? Would be greatly appreciated

**Comment by s-ayush2903** [comment]
Taking it up! :v:

**Comment by s-ayush2903** [comment]
* Since i made a clean install, I'm drown in the ocean of `loading...` that keeps on using my internet and does not do anything.
* Maybe i'll put a screen rec. here that demonstrates the aforementioned.
* In previous install it was working fine, cool.
* I'll update it as soon as i get my today's classes finished(~6:00pm IST)

**Comment by Frans-Lukas** [comment]
@s-ayush2903 is the `loading...` a bug or just your internet being annoying? Thanks for testing again.

**Comment by s-ayush2903** [comment]
Umm.. I can't claim this to be a bug as i couldn't find any file change that hit the networking layer of the module and this(loading continuing for a long time) always happens to me when i make a clean install of kiwix despite the internet speed is quite okayish ~200kbps. But still i'm attaching a screen cast mentioned in the previous comment regarding the same(connection speed)
* https://drive.google.com/file/d/13MxuEi_S-p--ZkVU595G-jQGIFGw1x-s/view?usp=sharing

I'll consider my ISP for this problem of connection as the changes reflected the same result as it was proposed and expected to be in both `app` and `custom` in cached installs

**Comment by Frans-Lukas** [comment]
@s-ayush2903 Seems to have been a bug. @macgills reverted a PR which fixed the issue. Thanks for finding it!

**Comment by macgills** [comment]
@s-ayush2903 I'll be awaiting your verification that this works before going ahead with a merge

**Comment by s-ayush2903** [comment]
After testing I would say: this PR does what it says but not always
* It works fine as expected 
* There is a common thing noticed when it doesn't work when user has expanded the sub-articles and get backs to them after exiting and starting the app, then all the sub-articles(sub-headings) are collapsed.

Check out the full demonstration video [right here](https://youtu.be/SUwkrCCdpGA)(with timestamps)
The behavior is same for both custom and app.

**Comment by s-ayush2903** [comment]
I think i cluttered up the previous comment or left it like a labyrinth, so here is the clear idea:
* It restores the state of tabs in the case when user does not leave the sub-articles open or leaves the app midway while reading a sub-article
* It could not perform as expected when user is reading a sub-article and leaves it open and retries opening the app. It occurs because while re-opening the application all the sub-articles are restored to their very initial state(the state in which they were by default added to the file).


* I think we need to save the state sub-articles too(if possible), this should do the trick


Q: What do i mean by sub-articles?
A: By sub-articles i'm referring to headings in the file that by default are present in collapsed state and on clicking show up the content in them resulting in detailed view

Q: What do i mean by very initial state?
A: I'm considering very initial state to be the state which is present when the file is first downloaded and that state of sub-articles

**Comment by Frans-Lukas** [comment]
Very nice demonstration! However, I think sub-article restoration is out of the scope for this ticket. @macgills what do you think?

**Comment by macgills** [comment]
Yes, I am pretty sure it has its own ticket already

**Review by macgills — APPROVED** [review]

**Comment by macgills** [comment]
Good work team!

**Comment by yyprince334** [comment]
I want to contribute in this

**Comment by gouri-panda** [comment]
@yyprince334 This is  a PR not a issue.

**Pull request merged**